data_prep/graph_dataset.py

```diff
@@ -1,72 +1,155 @@
 import abc
 
-from data_prep.dataset import Dataset
+import torch
 from torch_geometric.data import Dataset as GeometricDataset
+from torch_geometric.data import Data
+from transformers import RobertaTokenizerFast
+
+from data_prep.dataset import Dataset
+from data_prep.graph_utils import tf_idf_mtx, get_PMI
 
 
 class GraphDataset(Dataset, GeometricDataset):
     """
     Parent class for graph datasets.
     Require to implement a preprocess and generate_features methods.
     """
-    def __init__(self, *args, **kwargs):
-        # preprocess text
-        # compute tf.idf and pmi
-        # generate edges
-        # generate features
-        # build data object
-        raise NotImplementedError
+    def __init__(self, corpus, device):
+        self._device = device
+        self._tokenizer = RobertaTokenizerFast.from_pretrained("roberta-base")
+        self._num_classes = corpus.num_classes()
+
+        train_texts, train_labels = corpus.train_data()
+        val_texts, val_labels = corpus.val_data()
+        test_texts, test_labels = corpus.test_data()
+
+        self._raw_texts = train_texts + val_texts + test_texts
```

Comment:
for what again do we need the raw texts later? Or for what other reason are you do we store the actually inside this model

Reply:
Ok no, let me rephrase it: in `RobertaGraphDataset.__init__`, `self._raw_texts` is accessed as property... but then in the `_preprocess` you pass it as parameter... so this property is used in different ways in the same class. What is the motivation to once access it as a property but then pass it in the preprocess?

I would suggest to just also do `self._raw_texts`  in the preprocess and not pass it as parameter

Reply:
You are right, updated it